apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -1407,7 +1426,7 @@ def update_centroids():
                 config=config,
                 verbose=verbose,
                 trace_id=trace_id,
-            )
+            ).astype(np.float32)
```

Comment:
Is this needed even if we don't use sklearn?

Reply:
Not sure, I think that our vector to centroids assignments could handle vectors being different type than the centroids. However we need to test this.